Adyen/UI/View Controllers/Secured View Controller/SecuredViewController.swift

```diff
@@ -105,54 +97,42 @@ public final class SecuredViewController: UIViewController {
     }
     
     private func listenToBackgroundNotifications() {
-        let toBackgroundObserver = NotificationCenter.default.addObserver(forName: UIApplication.willResignActiveNotification,
-                                                                          object: nil,
-                                                                          queue: OperationQueue.main) { [weak self] _ in
-            self?.addBlur()
-        }
-        
-        let backFromBackgroundObserver = NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification,
-                                                                                object: nil,
-                                                                                queue: OperationQueue.main) { [weak self] _ in
-            self?.removeBlur()
-        }
-        
-        backgroundObservers = [toBackgroundObserver, backFromBackgroundObserver]
+        var array = [Any]()
+        array.append(notificationCenter.addObserver(forName: UIApplication.willResignActiveNotification,
+                                                    object: nil,
+                                                    queue: OperationQueue.main,
+                                                    using: { [weak self] _ in self?.addBlur() }))
+        array.append(notificationCenter.addObserver(forName: UIApplication.didBecomeActiveNotification,
+                                                    object: nil,
+                                                    queue: OperationQueue.main,
+                                                    using: { [weak self] _ in self?.removeBlur() }))
+        backgroundObservers = array
     }
     
     private func addBlur() {
         view.addSubview(blurEffectView)
-        blurEffectView.translatesAutoresizingMaskIntoConstraints = false
         view.backgroundColor = .clear
         
-        let constraints = [
-            blurEffectView.leftAnchor.constraint(equalTo: view.leftAnchor),
-            blurEffectView.topAnchor.constraint(equalTo: view.topAnchor),
-            blurEffectView.rightAnchor.constraint(equalTo: view.rightAnchor),
-            blurEffectView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
-        ]
-        
-        blurConstraints = constraints
-        
-        NSLayoutConstraint.activate(constraints)
+        blurEffectView.translatesAutoresizingMaskIntoConstraints = false
+        blurConstraints = blurEffectView.adyen.anchore(inside: view)
         
-        UIView.animate(withDuration: 0.2, animations: {
-            self.blurEffectView.effect = self.blurEffect
+        UIView.animate(withDuration: 0.2, animations: { [weak self] in
```

Comment:
not sure we need `[weak self]` here, since `self` does not retain UIView. same in a few other places.

Reply:
For me this is more a force of a habit..
Technically it does retain `self`, but only for 0.2 sec :D
I agree - most of the time `[weak self]` is unnecessary; in the other hand it is fail-safe - no harm in having it and it protects you from potential mistakes.

What do you think @mohammedDehairy ?
Should we use `[weak self]`:

[ ] ONLY when needed 
[ ] ALWAYS
[ ] without restrictions

Reply:
here its most probably not needed, but adding [weak self] doesn't hurt either, so no need to have a hard rule about it, I wouldn't mind adding it if its force of habit :-) .